### PR TITLE
Relaxing docker image source extended test

### LINKED ISSUE #19608: test flake: Timeout due to ruby version mismatch in docker source image tests
Extended test `[Feature:Builds][Slow] build can have Docker image source creating a build with an input source image and docker strategy` times out because underlying Docker build fails.

Root cause - ruby version mismatch leads to error running `scl enable rh-ruby22`

##### Version
v3.10.0-alpha0

##### Current Result
Test times out - underlying build errors.

##### Expected Result
Test passes

##### Additional Information
Test Log: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/19532/test_pull_request_origin_extended_builds/1572/

### PR DESCRIPTION
**Pull request by adambkaplan**
Only check if pod exists when examining build environment
Fixes #19608 (for real)

### CODE REVIEW

File: test/extended/builds/image_source.go [hunk 1]
```diff
@@ -251,3 +205,22 @@ var _ = g.Describe("[Feature:Builds][Slow] build can have Docker image source",
...
+func checkBuildEnv(value string) {
+	obj, _, err := legacyscheme.Codecs.UniversalDecoder().Decode([]byte(value), nil, nil)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	ok := false
+	build, ok := obj.(*buildapi.Build)
+	o.Expect(ok).To(o.BeTrue(), "could not convert build env\n %s\n to a build object", value)
+
+	o.Expect(build.Spec.Strategy.CustomStrategy.From.Kind).To(o.Equal("DockerImage"))
+	o.Expect(build.Spec.Strategy.CustomStrategy.From.Name).To(o.ContainSubstring("@sha256:"))
+	o.Expect(build.Spec.Strategy.CustomStrategy.PullSecret).NotTo(o.BeNil())
```
Comment (bparees) [inline test/extended/builds/image_source.go]: each test used a different field to check (build.Spec.Strategy.DockerStrategy vs SourceStrategy vs. CustomStrategy), so this won't work as a generic utility function.

**Comment by adambkaplan** [comment]
@bparees ptal (renamed a few of the pod check functions after @smarterclayton comments)

File: test/extended/util/framework.go [hunk 2]
```diff
@@ -1054,6 +1054,11 @@ var CheckPodIsReadyFn = func(pod kapiv1.Pod) bool {
 	return false
 }
 
+// CheckPodExists returns true if the pod exists or had existed
```
Comment (bparees) [inline test/extended/util/framework.go]: this godoc is misleading...   someone might read it and assume they could just invoke CheckPodExists(pod).  In reality it always returns true and can't be used standalone, it really only makes sense in the context of passing it as a no-op func to WaitForPods().

Comment (adambkaplan) [reply test/extended/util/framework.go]: I had considered updating this to do a more robust check (ex: does the pod have non-empty `ObjectMeta`)

Comment (bparees) [reply test/extended/util/framework.go]: just renaming it to CheckPodNoOp would make me happy.

**Comment by adambkaplan** [comment]
@bparees updated per feedback.

**Pull request merged**